### PR TITLE
Add edit a contact, create a contact, and get all projects for a contact endpoints

### LINKED ISSUE #15: Get all projects for a specific contact
Get all projects for a specific contact (query by the contact's UUID) 

Reason why we are using UUID – https://medium.com/@Mareks_082/auto-increment-keys-vs-uuid-a74d81f7476a

### LINKED ISSUE #23: Create a new contact
Create a new contact using the POST endpoint

Flask documentation on [Flask Blueprints](https://flask.palletsprojects.com/en/1.1.x/blueprints/#blueprints)
An example of the REST endpoints - [https://github.com/celinehuang/kom_get_your_bucha/blob/master/backend/app/items/views.py]()

### PR DESCRIPTION
**Pull request by dgiltz**
# Summary

Created contact creation and editing methods following flask architecture
	
## Related Issues

Which issues does this PR resolve/work on?
Closes #15 #23 #25

### CODE REVIEW

**Review by celinehuang — COMMENTED** [review]
This code doesn't run and can you please add tests?

test examples here: https://github.com/hack4impact-mcgill/mu-crm-tool/blob/109e8b4c02de661849bf92e0b5922693722912cd/backend/tests/test_project.py

instructions to run code in development environment here: https://github.com/hack4impact-mcgill/mu-crm-tool/blob/main/backend/README.md

**Comment by dgiltz** [comment]
Oh yea sorry I forgot to import abort from flask, so it should be good to go after that

File: backend/app/contact/routes.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+@contact.route("/<uuid:id>/", methods=["GET"])
+def get_all_projects(id):
+    contact = Contact.query.filter_by(id=id).first()
+    if contact is None:
+        abort(404, "No contact found with specified id")
+
+    # might just be overthinking this but this is the correct way to retrieve
+    # the data correct
+    data = contact.serialize
+    return data["projects"]
```
Comment (CurtisMIT) [inline backend/app/contact/routes.py:70]: @dgiltz If unsure you can write unit tests in the same pr (fill in the summary/test template that celine created) to test whether you're correct or not. I think you're on the right track but unfortunately the `Contact` model's `projects` attribute has not been added into its serialization.

File: backend/app/contact/routes.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+@contact.route("/<uuid:id>/edit", methods=["PUT"])
+def edit_contact(id):
+    contact = Contact.query.filter_by(id=id).first()
+    if contact is None:
+        abort(404, "No contact found with specified id")
+
+    db.session.delete(contact)
+    db.session.commit()
```
[the remark this answers is not in the document]
Comment (celinehuang) [reply backend/app/contact/routes.py:55]: I don't actually understand what this endpoint is doing, what are we editing here? there are no fields being passed in? and why are we deleting the contact

File: backend/app/contact/routes.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from flask import jsonify, request, abort
+import uuid
+from app import db
+from app.models import Contact
+from . import contact
+
+
+@contact.route("", methods=["POST"])
+def create_contact():
+    #
```
Comment (celinehuang) [inline backend/app/contact/routes.py]: remove empty comment

**Review by celinehuang — COMMENTED** [review]
Add the tests here to make sure CI is passing and your endpoints are working. As well as some code review changes :)

File: backend/app/contact/routes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (celinehuang) [reply backend/app/contact/routes.py:55]: No I feel like that's just unnecessary overhead if we are not changing anything when editing a contact we shouldn't allow the request to go through, unless you have other thoughts @CurtisMIT ?

[the remark this answers is not in the document]
Comment (CurtisMIT) [reply backend/app/contact/routes.py:55]: @celinehuang agree! Same as the project update route

**Comment by CurtisMIT** [comment]
Closing as @dgiltz has moved it to #115 and #119.

**Pull request closed without merging**